rclcpp/include/rclcpp/clock.hpp

```diff
@@ -78,6 +78,23 @@ class Clock
   Time
   now();
 
+  /**
+   * Sleep until a specified Time, according to clock type.
+   *
+   * Notes for RCL_ROS_TIME clock type:
+   *   - Can sleep forever if ros time is active and received clock never reaches `until`
+   *   - If ROS time enabled state changes during the sleep, this method will immediately return
+   *     false. There is not a consistent choice of sleeping time when the time source changes,
+   *     so this is up to the caller to call again if needed.
+   *
+   * \param until absolute time according to current clock type to sleep until
+   * \return true if the time `until` is reached
```

Comment:
I forgot this on the last review, but it might be nice to mention what happens when `until` is in the past (i.e. it returns true immediately).